libs/kotaemon/kotaemon/embeddings/fastembed.py

```diff
@@ -0,0 +1,68 @@
+from typing import TYPE_CHECKING, Optional
+
+from kotaemon.base import Document, DocumentWithEmbedding, Param
+
+from .base import BaseEmbeddings
+
+if TYPE_CHECKING:
+    from fastembed import TextEmbedding
+
+
+class FastEmbedEmbeddings(BaseEmbeddings):
+    """Utilize fastembed library for embeddings locally without GPU.
+
+    Supported model: https://qdrant.github.io/fastembed/examples/Supported_Models/
+    Code: https://github.com/qdrant/fastembed
+    """
+
+    model_name: str = Param(
+        "BAAI/bge-small-en-v1.5",
+        help=(
+            "Model name for fastembed. "
+            "Supported model: "
+            "https://qdrant.github.io/fastembed/examples/Supported_Models/"
+        ),
+    )
+    batch_size: int = Param(
+        256,
```

Comment:
If the lib actually does chunking and processes in batch, then this batch size might be a bit big for the average notebook.